server/src/main/java/org/opensearch/index/SegmentReplicationPressureService.java

```diff
@@ -201,6 +201,15 @@ public void setMaxAllowedStaleReplicas(double maxAllowedStaleReplicas) {
 
     public void setReplicationTimeLimitFailReplica(TimeValue replicationTimeLimitFailReplica) {
         this.replicationTimeLimitFailReplica = replicationTimeLimitFailReplica;
+        updateAsyncFailReplicaTask();
```

Comment:
I think the following will accomplish the same thing without any other changes? [code reference](https://github.com/opensearch-project/OpenSearch/blob/ad7ce4cd446523d52286530ccfbb1dabc2dc7f86/server/src/main/java/org/opensearch/common/util/concurrent/AbstractAsyncTask.java#L98-L116)

```suggestion
        this.replicationTimeLimitFailReplica = replicationTimeLimitFailReplica;
        if (TimeValue.ZERO.equals(replicationTimeLimitFailReplica)) {
            failStaleReplicaTask.setInterval(TimeValue.ZERO);
        } else {
            failStaleReplicaTask.setInterval(INTERVAL);
        }
```

Reply:
Actually, you'll have to remove the explicit call to `rescheduleIfNecessary()` and then ensure `setReplicationTimeLimitFailReplica()` gets called in the constructor as well.

Reply:
Yes this would also work - though we would also need to set:
```
        protected boolean mustReschedule() {
            return true;
        }
```
initially, or the task will get scheduled with a default value of zero.

Reply:
Made the update here to set interval instead of reconstructing the task.

Reply:
> Actually, you'll have to remove the explicit call to `rescheduleIfNecessary()` and then ensure `setReplicationTimeLimitFailReplica()` gets called in the constructor as well.

This won't work because `rescheduleIfNecessary` initializes the internal `cancellable` variable.  So without this `setInterval` will skip invoking `rescheduleIfNecessary`.